inst/examples/variable.cpp

```diff
@@ -38,6 +38,25 @@ class variable {
   dust::shared_ptr<variable> shared;
 };
 
+// This is required to activate gpu support. We can possibly do this
+// automatically based on the existance of the template below though.
+template <>
+struct dust::has_gpu_support<variable> : std::true_type {};
+
+template <>
+void update_device<variable>(size_t step,
+                             const dust::interleaved<variable::real_t> state,
+                             dust::interleaved<int> internal_int,
+                             dust::interleaved<variable::real_t> internal_real,
+                             dust::shared_ptr<variable> shared,
```

Comment:
I would suggest that this signature becomes:
```suggestion
                             real_t * shared_real,
                             int * shared_int,
```
But not sure this is possible in this PR

Reply:
The generation of these would be in `run_particles`

Reply:
yeah, that's not possible yet. Take a look [here](https://github.com/mrc-ide/sircovid/blob/master/inst/dust/basic.cpp#L72-L311) at the shared memory in the small sircovid basic model; we've got a motley collection of ints, reals and vectors of reals

Reply:
As discussed offline (well different online) we'll mock up some ideas around this starting with a single parameter set and see how we get on